lib/block/transaction_pool.go

```diff
@@ -0,0 +1,109 @@
+package block
+
+import (
+	"fmt"
+
+	"boscoin.io/sebak/lib/common"
+	"boscoin.io/sebak/lib/errors"
+	"boscoin.io/sebak/lib/storage"
+	"boscoin.io/sebak/lib/transaction"
+)
+
+// TransactionPool is not `transaction.Pool`; this contains the all the valid
+// transactions. The unconfirmed transactions of `TransactionPool` can be
+// removed, but confirmed transactions must be here.
+//
+// NOTE The sync process must store the confirmed transactions.
+type TransactionPool struct {
```

Comment:
@spikeekips Is `TransactionPool` for saving confirmed transactions before `finish Ballot`?

Reply:
No, it contains all the incoming transactions, of course they are well-formed. It will be used for

- storing incoming txs, finishBallot does not need to tx messages
- for syncer, api blocks will produce tx from TransactionPool